### PR TITLE
Update Calico manifests to v3.22.0

### LINKED ISSUE #1913: Update Calico manifests to latest version
Current Calico version in main is 3.19 https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/addons/calico.yaml, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/addons/calico-ipv6.yaml

Latest at time of writing this is 3.21.2 https://github.com/projectcalico/calico/releases.

We should update both manifests (default and ipv6) to use the latest version of Calico. This might require changes to the manifest itself, besides the version itself.

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**: 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1913 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update Calico manifests to v3.22.0
```

### CODE REVIEW

File: templates/addons/calico-ipv6.yaml [hunk 1]
```diff
@@ -314,6 +314,268 @@ status:
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
+metadata:
```
Comment (CecileRobertMichon) [inline templates/addons/calico-ipv6.yaml:317]: @jsturtevant these are all the changes made in the ipv6 manifest by the 3.20 -> 3.22 update. Let's see if this PR repros the issue that was happening in #2021

**Comment by CecileRobertMichon** [comment]
/retest

just in case, but pretty sure this is the same issue as last time

File: templates/addons/calico-ipv6.yaml (hunk 1 above)
Comment (jsturtevant) [reply templates/addons/calico-ipv6.yaml:317]: Interesting... I don't see anything that jumps out as different configuration

**Comment by jsturtevant** [comment]
> just in case, but pretty sure this is the same issue as last time

The previous error looks like the deployment pod didn't start:

```
        "reason": "MinimumReplicasUnavailable",
        "message": "Deployment does not have minimum availability."

LAST SEEN                      TYPE    REASON             OBJECT                MESSAGE
2022-02-25 19:49:10 +0000 UTC  Normal  ScalingReplicaSet  deployment/webgt66i4  Scaled up replica set webgt66i4-7887d46cd4 to 1
```

**Comment by jsturtevant** [comment]
found: 

```
Feb 25 19:52:14.155999 capz-e2e-ty3nms-ipv6-md-0-5c2fx kubelet[1104]: E0225 19:52:14.155716    1104 pod_workers.go:918] "Error syncing pod, skipping" err="failed to \"CreatePodSandbox\" for \"webgt66i4-7887d46cd4-xcnpg_default(0561d350-1d32-4ab2-85d3-2710a08d5f71)\" with CreatePodSandboxError: \"Failed to create sandbox for pod \\\"webgt66i4-7887d46cd4-xcnpg_default(0561d350-1d32-4ab2-85d3-2710a08d5f71)\\\": rpc error: code = Unknown desc = failed to setup network for sandbox \\\"8fa8ef3b00d43ebe98b1df30d9e9da958d86ede75403282adef971038dddcdc8\\\": plugin type=\\\"calico\\\" failed (add): usePodCidr found but there is no IPv4 CIDR configured\"" pod="default/webgt66i4-7887d46cd4-xcnpg" podUID=0561d350-1d32-4ab2-85d3-2710a08d5f71
Feb 25 19:52:26.177189 capz-e2e-ty3nms-ipv6-md-0-5c2fx kubelet[1104]: E0225 19:52:26.177083    1104 remote_runtime.go:116] "RunPodSandbox from runtime service failed" err="rpc error: code = Unknown desc = failed to setup network for sandbox \"86203b1ad4e9ea791ba34c6dfda6abeaad424e7461f8fa5d8be750d3a8f2bf5b\": plugin type=\"calico\" failed (add): usePodCidr found but there is no IPv4 CIDR configured"
```

which is configured here https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6a9300696954b36e032c26356d85bbb7487f32b5/templates/addons/calico-ipv6.yaml#L3609

**Comment by CecileRobertMichon** [comment]
Now seeing:
```
Feb 28 20:04:09.884471 capz-e2e-agi05s-ipv6-md-0-kr9lf kubelet[1142]: E0228 20:04:09.884377    1142 pod_workers.go:918] "Error syncing pod, skipping" err="failed to \"CreatePodSandbox\" for \"webvrg5ho-684d8f8b-g92ss_default(d4c7210c-4b95-4ec5-9e0a-e8f17130ae12)\" with CreatePodSandboxError: \"Failed to create sandbox for pod \\\"webvrg5ho-684d8f8b-g92ss_default(d4c7210c-4b95-4ec5-9e0a-e8f17130ae12)\\\": rpc error: code = Unknown desc = failed to setup network for sandbox \\\"8b20fbb36cbde9eb44363de2aa5fcf0544f33ebe599269396cdbe25bad50e047\\\": plugin type=\\\"calico\\\" failed (add): no IP ranges specified\"" pod="default/webvrg5ho-684d8f8b-g92ss" podUID=d4c7210c-4b95-4ec5-9e0a-e8f17130ae12
```

**Comment by CecileRobertMichon** [comment]
Looks like there is a `usePodCidrIPv6` option, trying that

**Comment by CecileRobertMichon** [comment]
This is the change related to this for the record: https://github.com/projectcalico/calico/pull/5313/files#diff-742c8968cd6af9cdd7510f39e61fe127ce4fa98fe893beb0183837182ec09072R340

File: templates/addons/calico/patches/remove-aws-reference.yaml [hunk 2]
```diff
@@ -1,5 +0,0 @@
-# Drops the FELIX_AWSSRCDSTCHECK field
-# json patch doesn't allow for dropping fields in array by name
-# https://github.com/projectcalico/calico/issues/5101
-- op: remove
-  path: "/spec/template/spec/containers/0/env/12"
```
Comment (jsturtevant) [inline templates/addons/calico/patches/remove-aws-reference.yaml:5]: nice that we get to drop this!  fyi @jackfrancis

**Review by jsturtevant — COMMENTED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
> Init container clonerefs not ready: (state: waiting, reason: "PodInitializing", message: "") Init container initupload not ready: (state: waiting, reason: "PodInitializing", message: "") Init container place-entrypoint not ready: (state: waiting, reason: "PodInitializing", message: "")

prow pod timeout 

/test pull-cluster-api-provider-azure-windows-containerd-upstream-with-ci-artifacts

**Comment by CecileRobertMichon** [comment]
all tests passed

/approve

**Pull request merged**